beetmoverscript/utils.py

```diff
@@ -286,3 +287,11 @@ def get_bucket_name(context, product):
 
 def get_bucket_url_prefix(context):
     return context.config['bucket_config'][context.bucket]['url_prefix']
+
+
+def validated_task_id(task_id):
+    """Validate the format of a taskcluster taskId."""
+    pattern = r'^[A-Za-z0-9_-]{8}[Q-T][A-Za-z0-9_-][CGKOSWaeimquy26-][A-Za-z0-9_-]{10}[AQgw]$'
```

Comment:
I'm wondering if it wouldn't make more sense here to reuse `taskcluster` python package. There must be something there already for validating taskIds. Or maybe even query the queue for the existence of that task? Seems a bit less fragile than a regex that may change in the future with more Taskcluster development. Certainly not blocking now though, just a nit.

Reply:
I don't see that feature in taskcluster-client.py at the moment. Some of their tests use taskIds like `123`, so I'd imagine there's no internal validation in the module, too. 

We could ask them to add it, of course - the regex was taken from https://docs.taskcluster.net/docs/reference/platform/taskcluster-queue/references/api - we could query https://schemas.taskcluster.net/queue/v1/task.json# for the live value if nothing ends up being added to the client.